app/jobs/mailers/monthly_stats.rb

```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Mailers
+  class MonthlyStats < ApplicationJob
+    queue_as :mailers
+
+    def perform
+      User.where(provider: 'azure_oauth', opt_out_stats_email: false, active: true).each do |user|
```

Comment:
Generally speaking, always use `.find_each` instead of `.each` when dealing with ActiveRecord. What that does is secretly  under the hood load records out in batches, rather than attempting to load out the entire result set at once (which can lead to memory swapping).